staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/azure.go

```diff
@@ -78,17 +80,29 @@ func (c *azureTokenCache) setToken(tokenKey string, token *azureToken) {
 }
 
 func newAzureAuthProvider(_ string, cfg map[string]string, persister restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
-	var ts tokenSource
-
-	environment, err := azure.EnvironmentFromName(cfg[cfgEnvironment])
+	var (
+		ts                 tokenSource
+		omitTokenSPNPrefix bool
+		environment        azure.Environment
+		err                error
+	)
+
+	environment, err = azure.EnvironmentFromName(cfg[cfgEnvironment])
 	if err != nil {
 		environment = azure.PublicCloud
 	}
-	ts, err = newAzureTokenSourceDeviceCode(environment, cfg[cfgClientID], cfg[cfgTenantID], cfg[cfgApiserverID])
+
+	if cfg[cfgOmitTokenSPNPrefix] != "" {
+		omitTokenSPNPrefix, err = strconv.ParseBool(cfg[cfgOmitTokenSPNPrefix])
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s", cfgOmitTokenSPNPrefix)
```

Comment:
@weinong can we also add the error to the message?

Reply:
make sense